src/opm/parser/eclipse/share/keywords/000_Eclipse100/R/ROCK

```diff
@@ -3,7 +3,7 @@
   "sections": [
     "PROPS"
   ],
-  "size": "UNKNOWN",
+    "size": {"keyword" : "TABDIMS", "item" : "NTPVT"},
```

Comment:
I don't think this is completely correct.  Using `NTPVT` for the number of records **is** correct if `ROCKOPTS` is **not** present (common case), but if the run sets `ROCKOPTS` before `ROCK` then the number of `ROCK` records is defined by the following relations
1. If `ROCKOPTS(3) = 'SATNUM'` then the number of records is `NTSFUN (= TABDIMS(1))`
2. If `ROCKOPTS(3) = 'ROCKNUM'` then the number of records is `NTROCC (= TABDIMS(13))`
3. If `ROCKOPTS(3) = 'PVTNUM'` (default setting) then the number records is `NTPVT (= TABDIMS(2))`

And as a final wrinkle, if `ROCKOPTS(3) = 'ROCKNUM'` but `NTROCC` is defaulted then we should also use `NTPVT`.  I don't know how to model these interactions in JSON.

Reply:
> Using NTPVT for the number of records is correct if ROCKOPTS is not present (common case), but if the run sets ROCKOPTS before ROCK then the number of ROCK records is defined by the following relations [...]

Thank you - I had no idea.

> I don't know how to model these interactions in JSON.

I'll go on stage and say it is impossible with the correct scheme; I'll revert back to `"size" : "UNKNOWN"` and add your detailed explanation as a comment.